src/Imagine/Filter/FilterManager.php

```diff
@@ -79,7 +79,7 @@ public function apply(BinaryInterface $binary, array $config): BinaryInterface
         return $this->applyPostProcessors($this->applyFilters($binary, $config), $config);
     }
 
-    public function applyFilters(BinaryInterface $binary, array $config): BinaryInterface
+    public function applyFilters(BinaryInterface $binary, array $config): Binary
```

Comment:
i'd rather keep this at the BinaryInterface. consumers should not rely on the specific implementation, to stay flexible.

if for some reason we would later want to have other binary implementations (e.g. for streaming, or whatever) it would be a BC break to change. if we only declare the interface to be returned, we are more flexible.